### PR TITLE
Adding github actions file to build and push docker image to docker hub / ECR

### PR DESCRIPTION
**Pull request by adityabharadwaj198**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
*  Refactoring

* **What is the current behavior?** (You can also link to an open issue here)
Currently the same github action exists in the open_source_CI repo (https://github.com/marqo-ai/open_source_CI/blob/main/.github/workflows/build_push_img.yml). We want to move it here.

* **What is the new behavior (if this is a feature change)?**
Behavior remains the same - the action can now be invoked from main Marqo repo

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
No

* **Related Python client changes** (link commit/PR here)
NA

* **Related documentation changes** (link commit/PR here)
NA

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: .github/workflows/build_push_img.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_EC2_GH_RUNNER_ACCESS_KEY_ID }}
```
Comment (wanliAlex) [inline .github/workflows/build_push_img.yml]: What is the AWS account? OpenSource or S2Search Core?

File: .github/workflows/build_push_img.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      marqo_ref:
+        description: 'Marqo branch-name, commit SHA or tag'
+        required: false
+        default: 'mainline'
+      push_to:
+        description: 'image destination. Options: "ECR", "DockerHub"'
```
Comment (wanliAlex) [inline .github/workflows/build_push_img.yml:11]: Make them drop down options?

**Review by wanliAlex — CHANGES_REQUESTED** [review]

File: .github/workflows/build_push_img.yml (hunk 2 above)
Comment (adityabharadwaj198) [reply .github/workflows/build_push_img.yml:11]: Good suggestion, Sure!

Comment (adityabharadwaj198) [reply .github/workflows/build_push_img.yml:11]: done

File: .github/workflows/build_push_img.yml (hunk 1 above)
Comment (adityabharadwaj198) [reply .github/workflows/build_push_img.yml]: This is open source now - I changed the secrets to use opensource

File: .github/workflows/build_push_img.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.MARQO_WORKFLOW_TESTS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.MARQO_WORKFLOW_TESTS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (wanliAlex) [inline .github/workflows/build_push_img.yml:44]: Why do we start the instance on us-west-2 here? Should it also be on us-east-1?

File: .github/workflows/build_push_img.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          # 200 GB amd64 image
+          ec2-image-id:  ami-01accacd51bdde263
+          ec2-instance-type: t3.xlarge
+          subnet-id: subnet-038b1447ac3c97e7a
+          security-group-id: sg-094be521399e0d5ba
```
Comment (wanliAlex) [inline .github/workflows/build_push_img.yml:55]: Maybe add some tags like this? https://github.com/marqo-ai/marqo/blob/5c372ddc48498a50ea1fb8f53364644f6fbefcaa/.github/workflows/unit_test_200gb_CI.yml#L47-L55

**Review by wanliAlex — CHANGES_REQUESTED** [review]

File: .github/workflows/build_push_img.yml (hunk 4 above)
Comment (adityabharadwaj198) [reply .github/workflows/build_push_img.yml:55]: Good idea, done.

**Review by wanliAlex — APPROVED** [review]

**Pull request merged**